File: ambeth/Ambeth.DataChange/ambeth/datachange/model/DirectDataChangeEntry.cs
using System;
using De.Osthus.Ambeth.Datachange.Model;

namespace De.Osthus.Ambeth.Datachange.Model
{
    public class DirectDataChangeEntry : IDataChangeEntry
    {
        public Object Entry { get; protected set; }

        public DirectDataChangeEntry(Object entry)
        {
            this.Entry = entry;
        }

        public Type EntityType
        {
            get
            {
                return Entry.GetType();
            }
        }

        public Object Id
        {
            get
            {
                return null;
            }
        }

        public sbyte IdNameIndex
        {
            get
            {
                return -1;
            }
        }

        public Object Version
        {
            get
            {
                return null;
            }
        }

        public String[] Topics
        {
            get
            {
                return null;
            }
            set
            {
            }
        }
    }
}

File: jambeth/jambeth-datachange/src/main/java/com/koch/ambeth/datachange/model/DirectDataChangeEntry.java
package com.koch.ambeth.datachange.model;

/*-
 * #%L
 * jambeth-datachange
 * %%
 * Copyright (C) 2017 Koch Softwaredevelopment
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

     http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
 * #L%
 */

public class DirectDataChangeEntry implements IDataChangeEntry
{
	protected Object entry;

	public DirectDataChangeEntry(Object entry)
	{
		this.entry = entry;
	}

	public Object getEntry()
	{
		return entry;
	}

	@Override
	public Class<?> getEntityType()
	{
		return this.entry.getClass();
	}

	@Override
	public Object getId()
	{
		return null;
	}

	@Override
	public byte getIdNameIndex()
	{
		return -1;
	}

	@Override
	public Object getVersion()
	{
		return null;
	}

	@Override
	public String[] getTopics()
	{
		return null;
	}

	@Override
	public void setTopics(String[] topics)
	{
	}
}
